tools/querytee/response_comparator.go

```diff
@@ -0,0 +1,208 @@
+package querytee
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+// ResponsesComparatorFunc helps with comparing different responses from various routes.
+type ResponsesComparatorFunc func(expected, actual []byte) error
```

Comment:
No need to add `Func` to the name.

Reply:
This type should be defined in `proxy_endpoint.go` where it is actually used.

I would (weakly) prefer to use interface for this, it's typically easier to find implementations for interface than for func type.